File: rust_src/70.爬楼梯.rs
/*
 * @lc app=leetcode.cn id=70 lang=rust
 *
 * [70] 爬楼梯
 */

// @lc code=start
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
		if n == 1
		{
			return 1;
		}
		if n == 2
		{
			return 2;
		}

		let mut dp0 = 1;
		let mut dp1 = 2;

		for i in 2..n
		{
			dp1 = dp1 + dp0;
			dp0 = dp1 - dp0;
		}

		return dp1;
    }
}
// @lc code=end

File: python_src/70.爬楼梯.py
#
# @lc app=leetcode.cn id=70 lang=python3
#
# [70] 爬楼梯
#
from typing import *

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        dp0 = 1
        dp1 = 1

        for _ in range(2, n + 1):
            # dp[i] = dp[i - 1] + dp[i - 2]

            dp = dp0 + dp1
            dp0 = dp1
            dp1 = dp

        return dp1
# @lc code=end

s = Solution()
print([s.climbStairs(i) for i in range(1, 46)])
